### PR TITLE
feat(inputs.rabbitmq): Add type tag to queues

### LINKED ISSUE #17632: Add queue type to rabbitmq queues
### Use Case

Add the ability to differentiate queues by type 

### Expected behavior

Queues can be differentiated by type

### Actual behavior

no way to get queue type

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by Goorsky1**
## Summary
Add type tag to queue to be able to differentiate queues by type 

## Checklist
- [x] No AI generated code was used in this PR

## Related issues

resolves #17632

### CODE REVIEW

**Comment by skartikey** [comment]
@Goorsky1 Thanks for the PR!

 Potential Issues/Suggestions
  1. The Type field is extracted directly without validation. Consider handling cases where type might be empty or unexpected.
  2. Only tests "classic" type. Should test other queue types (quorum, stream) if RabbitMQ supports them.
  3. No mention of minimum RabbitMQ version required for this field. Older versions might not provide the type field.
  4. If type is empty/missing, consider defaulting to "classic" for backward compatibility with older RabbitMQ versions.

**Comment by Goorsky1** [comment]
> @Goorsky1 Thanks for the PR!
> 
> Potential Issues/Suggestions
> 
> 1. The Type field is extracted directly without validation. Consider handling cases where type might be empty or unexpected.
> 2. Only tests "classic" type. Should test other queue types (quorum, stream) if RabbitMQ supports them.
> 3. No mention of minimum RabbitMQ version required for this field. Older versions might not provide the type field.
> 4. If type is empty/missing, consider defaulting to "classic" for backward compatibility with older RabbitMQ versions.

thanks for feedback!
I added validation for the unexpected or missing value for queue type, in both cases it defaults to "classic" 
Also I expanded the test to test for quorum,stream,unexpected and empty type. 

Let me know if you have any other suggestions.

File: plugins/inputs/rabbitmq/rabbitmq.go [hunk 1]
```diff
@@ -556,13 +573,25 @@ func gatherQueues(r *RabbitMQ, acc telegraf.Accumulator) {
...
+
+		queueType := queue.Type
+		if queueType == "" {
+			queueType = queueTypeClassic // default to classic for backward compatibility
+		}
+
+		if !isValidQueueType(queueType) {
+			r.Log.Warnf("Queue %q has unknown type '%s', defaulting to 'classic'", queue.Name, queueType)
+			queueType = queueTypeClassic
+		}
```
Comment (srebhan) [inline plugins/inputs/rabbitmq/rabbitmq.go]: Why not simply add the queue-type as a tag without any validation?

**Review by srebhan — CHANGES_REQUESTED** [review]
Thanks @Goorsky1 for your contribution! 

While I see no issues with adding the queue-type as a tag we cannot do this by default as we generate new series and in turn increase cardinality and potentially break existing queries. So please add a setting to the configuration to __enable__ adding the queue-type tag with default being not to add the tag!
Additionally I would not validate the queue type but simply default to `classic` and override if the type is returned with the message...

**Comment by Goorsky1** [comment]
> Thanks @Goorsky1 for your contribution!
> 
> While I see no issues with adding the queue-type as a tag we cannot do this by default as we generate new series and in turn increase cardinality and potentially break existing queries. So please add a setting to the configuration to **enable** adding the queue-type tag with default being not to add the tag! Additionally I would not validate the queue type but simply default to `classic` and override if the type is returned with the message...

Thanks for feedback! 

I made the queue type tag a opt in feature and made it default to classic when empty otherwise using the value returner by rabbit

File: plugins/inputs/rabbitmq/rabbitmq.go [hunk 2]
```diff
@@ -565,6 +573,13 @@ func gatherQueues(r *RabbitMQ, acc telegraf.Accumulator) {
 			"auto_delete": strconv.FormatBool(queue.AutoDelete),
 		}
 
+		if r.IncludeQueueTypeTag {
+			if queue.Type == "" {
+				queue.Type = queueTypeClassic // default to classic
+			}
+			tags["type"] = queue.Type
+		}
+
```
Comment (srebhan) [inline plugins/inputs/rabbitmq/rabbitmq.go:579]: How about

```suggestion
		if r.IncludeQueueTypeTag {
			if queue.Type == "" {
				tags["type"] = "classic"
			} else {
				tags["type"] = queue.Type
			}
		}

```

Comment (Goorsky1) [reply plugins/inputs/rabbitmq/rabbitmq.go:579]: Thanks again! I kept the old tests behavoiur unchanged and deleted the const
